test/test_interpreter.cpp

```diff
@@ -49,62 +49,44 @@ namespace test_kernel
         if (code.compare("?") == 0)
         {
             std::string html_content = R"(<iframe class="xpyt-iframe-pager" src="
-                https://xeus.readthedocs.io"></iframe>)";
-
-            kernel_res["status"] = "ok";
-            kernel_res["payload"] = nl::json::array();
-            kernel_res["payload"][0] = nl::json::object({
-                {"data", {
-                    {"text/plain", "https://xeus.readthedocs.io"},
-                    {"text/html", html_content}}
-                },
-                {"source", "page"},
-                {"start", 0}
-            });
-            kernel_res["user_expressions"] = nl::json::object();
-
-            return kernel_res;
+                                            https://xeus.readthedocs.io"></iframe>)";
+
+            auto payload = nl::json::array();
+            payload[0] = nl::json::object({
+                            {"data", {
+                                {"text/plain", "https://xeus.readthedocs.io"},
+                                {"text/html", html_content}}
+                            },
+                            {"source", "page"},
+                            {"start", 0}
+                        });
+
+            return xeus::create_successful_reply(payload);
```

Comment:
You may want to move the payload to avoid costly copy.

Reply:
What do you mean with move @JohanMabille ? Sorry, I don't get it.
I don't think it makes a lot of difference in this case because it's just an example?

Reply:
I was thinking of something like `return xeus::create_successful_reply(std::move(payload))` but that won't work given the signature of `xeus::create_successful_reply`, so forget my remark.

> I don't think it makes a lot of difference in this case because it's just an example?

A lot of people read examples to understand how to use the library; so even when it does not make a big difference, it is always worth writing the examples as you want the users to use the library.